drivers/rf/cc1101/cc1101.c

```diff
@@ -0,0 +1,1077 @@
+		return ret;
+	}
+
+	return ret;
+}
+
+int cc1101_reg_write(const struct device *dev,
+		uint8_t addr,
+		uint8_t val)
+{
+	int ret;
+	const struct cc1101_config *cfg = dev->config;
+	uint8_t cmd[] = { (addr | CC1101_SPI_WRITE), val };
+	const struct spi_buf tx_buf = {
+		.buf = cmd,
+		.len = sizeof(cmd)
+	};
+	const struct spi_buf_set tx = {
+		.buffers = &tx_buf,
+		.count = 1
+	};
+	struct cc1101_data *data = dev->data;
+	const struct spi_buf rx_buf = {
+		.buf = &data->status.reg,
+		.len = 1
+	};
+	const struct spi_buf_set rx = {
+		.buffers = &rx_buf,
+		.count = 1
+	};
+
+	ret = spi_transceive_dt(&cfg->spi, &tx, &rx);
+	if (ret) {
+		LOG_ERR("SPI write failed: %d\n", ret);
+		return ret;
+	}
+	return ret;
+}
+
+
+int cc1101_burst_read(const struct device *dev,
+		uint8_t start,
+		uint8_t *buf,
+		int size)
+{
+	int ret;
+	uint8_t addr;
+	const struct cc1101_config *cfg = dev->config;
+	struct cc1101_data *data = dev->data;
+	const struct spi_buf tx_buf = {
+		.buf = &addr,
+		.len = 1
+	};
+	const struct spi_buf_set tx = {
+		.buffers = &tx_buf,
+		.count = 1
+	};
+	struct spi_buf rx_buf[2];
+	const struct spi_buf_set rx = {
+		.buffers = rx_buf,
+		.count = ARRAY_SIZE(rx_buf)
+	};
+
+	addr = start | CC1101_SPI_BURST_READ;
+
+	rx_buf[0].buf = &data->status.reg;
+	rx_buf[0].len = 1;
+
+	rx_buf[1].buf = buf;
+	rx_buf[1].len = size;
+
+	ret = spi_transceive_dt(&cfg->spi, &tx, &rx);
+	if (ret) {
+		LOG_ERR("SPI burst read failed: %d\n", ret);
+		return ret;
+	}
+
+	return ret;
+}
+
+int cc1101_burst_write(const struct device *dev,
+		uint8_t start,
+		uint8_t *buf,
+		int size)
+{
+	int ret;
+	uint8_t addr;
+	const struct cc1101_config *cfg = dev->config;
+	struct spi_buf tx_buf[2];
+	const struct spi_buf_set tx = {
+		.buffers = tx_buf,
+		.count = ARRAY_SIZE(tx_buf)
+	};
+	struct cc1101_data *data = dev->data;
+	const struct spi_buf rx_buf = {
+		.buf = &data->status.reg,
+		.len = 1
+	};
+	const struct spi_buf_set rx = {
+		.buffers = &rx_buf,
+		.count = 1
+	};
+
+	addr = start | CC1101_SPI_BURST_WRITE;
+
+	tx_buf[0].buf = &addr;
+	tx_buf[0].len = 1;
+
+	tx_buf[1].buf = buf;
+	tx_buf[1].len = size;
+
+	ret = spi_transceive_dt(&cfg->spi, &tx, &rx);
+	if (ret) {
+		LOG_ERR("SPI burst write failed: %d\n", ret);
+		return ret;
+	}
+	return ret;
+}
+
+int cc1101_strobe(const struct device *dev, uint8_t addr)
+{
+	int ret;
+	const struct cc1101_config *cfg = dev->config;
+	uint8_t strobe[] = { (addr | CC1101_SPI_WRITE) };
+	const struct spi_buf tx_buf = {
+		.buf = strobe,
+		.len = sizeof(strobe)
+	};
+	const struct spi_buf_set tx = {
+		.buffers = &tx_buf,
+		.count = 1
+	};
+	struct cc1101_data *data = dev->data;
+	const struct spi_buf rx_buf = {
+		.buf = &data->status.reg,
+		.len = 1
+	};
+	const struct spi_buf_set rx = {
+		.buffers = &rx_buf,
+		.count = 1
+	};
+
+	ret = spi_transceive_dt(&cfg->spi, &tx, &rx);
+	if (ret) {
+		LOG_ERR("SPI strobe failed: %d\n", ret);
+		return ret;
+	}
+	return ret;
+}
```

Comment:
Looks like you could reduce the code size above by having one generic function for accessing SPI and having helpers for just reg access on top. See cc1200 or cc2520 driver code for instance